test/unit/application_helper_test.rb

```diff
@@ -28,4 +28,69 @@ def controller
       is_active_tab?('invalid_tab')
     end
   end
+
+  # Tests for active_announcements method
+  def setup_announcements
+    @user = users(:founder)
+    # Set up some mock announcements
+    mock_announcements = [
+      { 'version' => 1, 'message' => 'First announcement' },
+      { 'version' => 2, 'message' => 'Second announcement' },
+      { 'version' => 3, 'message' => 'Third announcement' }
+    ]
+    Announcements.instance_variable_set(:@announcements, mock_announcements)
+  end
+
+  def teardown_announcements
+    # Clear memoized announcements so tests remain isolated
+    Announcements.instance_variable_set(:@announcements, nil)
+  end
+
+  test 'active_announcements returns all announcements when no user' do
+    setup_announcements
+    
+    # Simulate no current user
+    def current_user
+      nil
+    end
+
+    active = active_announcements
+    assert_equal 3, active.length
+    assert_equal [1, 2, 3], active.map { |a| a['version'] }
+    
+    teardown_announcements
+  end
+
+  test 'active_announcements filters dismissed announcements for logged in user' do
+    setup_announcements
+    
+    # Dismiss announcement version 2
+    DismissedAnnouncement.create!(user: @user, announcement_version: 2)
+
+    # Simulate current user
+    def current_user
+      @user
+    end
+
+    active = active_announcements
+    assert_equal 2, active.length
+    assert_equal [1, 3], active.map { |a| a['version'] }
+    
+    teardown_announcements
+  end
+
+  test 'active_announcements returns all when user has no dismissed announcements' do
+    setup_announcements
+    
+    # Simulate current user with no dismissed announcements
+    def current_user
+      @user
+    end
+
+    active = active_announcements
+    assert_equal 3, active.length
+    assert_equal [1, 2, 3], active.map { |a| a['version'] }
+    
+    teardown_announcements
+  end
 end
```

Comment:
These tests are not good, don’t populate the application helper. Follow the example of other helpers where they act as them